packages/wdio-browserstack-service/src/service.ts

```diff
@@ -111,64 +144,113 @@ export default class BrowserstackService implements Services.ServiceInstance {
-        if (isAuthenticated && isHealingEnabled) {
-            const sessionToken = authInfo.sessionToken
-            await aiSDK.BrowserstackHealing.setToken(this._browser.sessionId, sessionToken, TCG_URL)
-
-            const browser:any = this._browser as WebdriverIO.Browser
-            browser.overwriteCommand('findElement', async (orginalFunc: (arg0: string, arg1: string) => any, using: string, value: string) => {
-                const sessionId = browser.sessionId
-                const tcgInfo = {
-                    tcgRegion: 'use',
-                    tcgUrl: TCG_URL,
+
+            aiSDK.AISDK.configure({
+                domain: 'https://tcg-preprod.bsstag.com',
+                platform: 'desktop',
+            })
+
+            function getFrameworkImpl(browser: any): NLToSteps.NLToStepsFrameworkImpl {
+                return {
+                    executeScript: async (script: (...data: any) => any, args: any[]) => {
+                        return await browser.execute(script, ...args)
+                    },
+                    getBrowser() {
+                        return browser.capabilities.browserName || 'chrome'
+                    }
                 }
-                let tcgDetails = `{"region": "${tcgInfo.tcgRegion}", "tcgUrls": {"${tcgInfo.tcgRegion}": {"endpoint": "${tcgInfo.tcgUrl.split('s://')[1]}"}}}`
-                tcgDetails = tcgDetails.replace(/'/g, '\\\'').replace(/"/g, '\\"')
-                const locatorType = using.replace(/'/g, '\\\'').replace(/"/g, '\\"')
-                const locatorValue = value.replace(/'/g, '\\\'').replace(/"/g, '\\"')
+            }
+
+            async function testNLToStepsStart() {
                 try {
-                    const result = await orginalFunc(using, value)
-                    if (!result.error) {
-                        const script = await aiSDK.BrowserstackHealing.logData(locatorType, locatorValue, undefined, undefined, authInfo.groupId, sessionId, undefined, tcgDetails)
-                        if (script) {
-                            await browser.execute(script)
-                        }
-                        return result
+                    const out = await NLToSteps.start({
+                        id: 'devqa-' + uuidv4(),
+                        objective: userInput,
+                        waitCallback: async (waitAction) => {
+                            console.log('Wait action called:', waitAction)
+                            return { waitAction, done: true }
+                        },
+                        authMethod: getAuthToken,
+                        frameworkImplementation: getFrameworkImpl(browser)
+                    })
+
+                    console.log('NLToSteps.start output:', out)
+                } catch (error) {
+                    console.error('Error in NLToSteps.start:', error)
+                }
+            }
+
+            await testNLToStepsStart()
```

Comment:
Should we wrap the `Error` object this would throw with a user-friendly error on what happened after `browser.ai("...")` call? Looks like stacktrace generated would be fine here, but please verify